src/components/WelcomeMat/WelcomeMat.js

```diff
@@ -0,0 +1,165 @@
+                    {title}
+                  </h2>
+                  <div className="slds-welcome-mat__info-description slds-text-longform">
+                    {description}
+                  </div>
+                  {renderProgress && !isSplash && renderProgress({ steps })}
+                  {(renderAction || renderDismiss) && (
+                    <div className="slds-welcome-mat__info-actions">
+                      {renderAction && renderAction({ onClose, steps })}
+                      {renderDismiss && (
+                        renderAction
+                          ? (
+                            <div className="slds-m-top_large">
+                              {renderDismiss({ id, onClose, steps })}
+                            </div>
+                          ) : renderDismiss({ id, onClose, steps })
+                      )}
+                    </div>
+                  )}
+                </div>
+              </Column>
+              {!isSplash && (
+                <Column
+                  as="ul"
+                  className={cx([
+                    'slds-welcome-mat__tiles',
+                    { 'slds-welcome-mat__tiles_info-only': isInfoOnly },
+                  ])}
+                  omitCol
+                  size="1-of-2"
+                >
+                  {steps.map(step => (
+                    <WelcomeMatStep
+                      key={`${id}-${step.id}`}
+                      isInfoOnly={isInfoOnly}
+                      onComplete={this.onComplete}
+                      step={step}
+                    />
+                  ))}
+                </Column>
+              )}
+            </Grid>
+          </div>
+        </ModalContent>
+      </Modal>
+    );
+  }
+}
+WelcomeMat.defaultProps = {
+  isInfoOnly: false,
+  isOpen: false,
+  onCompleteStep: null,
+  renderAction: null,
+  renderDismiss: null,
+  renderProgress: null,
+};
+
+WelcomeMat.propTypes = {
+  /**
+   * Description, rendered below `title`. Should return a `p`
+   */
+  description: PropTypes.node.isRequired,
+  /**
+   * Identifier for `Modal`
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Controls `isOpen` state of the underlying `Modal`
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * When this flag is set, `isComplete` is ignored for steps
+   */
+  isInfoOnly: PropTypes.bool,
+  /**
+   * Passed to `Modal`
+   */
+  onClose: PropTypes.func.isRequired,
+  /**
+   * Handler injected when rendering steps. Called with `({ id, step })`
+   */
+  onCompleteStep: PropTypes.func,
+  renderAction: PropTypes.func,
+  /**
+   * Should render a `Checkbox` or similar. Called with `({ onClose, steps })`
+   */
+  renderDismiss: PropTypes.func,
+  /**
+   * Should render a `WelcomeMatProgress` element. Called with `({ steps })`
+   */
+  renderProgress: PropTypes.func,
+  /**
+   * Steps. If set to `false` the "Splash" variant will be rendered
+   */
+  steps: PropTypes.oneOfType([
+    PropTypes.oneOf([false]),
```

Comment:
That's what I don't like most about JS. Passing absolutely different things to one component property or to one function argument.
I'd prefer to have splash variant in case steps are not set.

Reply:
Hm it depends, overloading a prop/argument can sometimes make sense. I think in this case, overloading steps with `false` is clearer than `undefined`. `steps={[]}` and `steps={undefined}` should render the same to avoid bad surprises. Setting `steps=false` is more expressive than passing `steps={/*ignored&*/} isSplash` imo, but that would be another option that we could think about.

Reply:
IMHO not giving `steps` prop to the component at all is more clear way to say that we want splash. And `undefined` and empty array are also different things, and so should be treated and rendered differently.
Also `steps={false}` to me looks like:
1) violation of DOM standard prop passing.
2) violation of semantics. Wording "steps" means we expect a collection of something. And `false` is not a collection.